src/client/components/stage-details/stage-details.component.js

```diff
@@ -0,0 +1,171 @@
+import React, { useState, useEffect } from 'react';
+import PropTypes from 'prop-types';
+import './stage-details.style.css';
+import SectionHeader from '../section-header/section-header.component';
+import Button from '../button/button.component';
+
+function capitalizeString(str) {
+  return str[0].toUpperCase().concat(str.slice(1, str.length));
+}
+
+export default function StageDetails({
+  cropID,
+  parametersAndLabels,
+  stages,
+  levels,
+  data,
+}) {
+  // Get the parameter data for crop with id of cropID
+  const [cropData, setCropData] = useState([]);
+
+  useEffect(() => {
+    setCropData(
+      data.filter((dataPoint) => parseInt(dataPoint.fk_crop_id, 10) === cropID),
+    );
+  }, [cropID, data]);
+
+  // Select the last stage to display as default
+  const [selectedStageNumber, setSelectedStageNumber] = useState(
+    Math.max(...stages.map((s) => s.id)),
+  );
+
+  // Get data for the currently selected stage
+  const [dataForStage, setDataForStage] = useState([]);
+
+  useEffect(() => {
+    setDataForStage(
+      cropData
+        ? cropData.filter(
+            (dataPoint) =>
+              parseInt(dataPoint.fk_crop_stage_id, 10) === selectedStageNumber,
+          )
+        : [],
+    );
+  }, [selectedStageNumber, cropData]);
+
+  function handleInputChange(e, level, parameter) {
```

Comment:
`parameter` seems very generic, could basically be anything? could we make it more descriptive 🤔

Reply:
"parameter" is the name Seasony use in their DB ([diagram](https://dbdiagram.io/d/5e79e58e4495b02c3b88b043)) for their measures (temperature, water_level, EC, PH, humidity), so I thought this would be the clearest way of naming it. 🤔

Reply:
Alright, if that makes sense to you then that is good 👍 I can live with that, I am just not a fan of generic namings but if the whole picture makes sense let's keep it as is 👍